### PR TITLE
Add niftyswap dev-dependency

### PR DESCRIPTION
**Pull request by xiam**
This is required for https://github.com/horizon-games/issue-tracker/issues/5306

### CODE REVIEW

**Comment by pkieltyka** [comment]
btw, how come we need this in go-sequence?

**Comment by xiam** [comment]
I understood this from the description here: https://github.com/horizon-games/issue-tracker/issues/5306, this note in particular:

> ..However, one thing is the "contracts" package which we use to get go bindings to a contract is in the go-sequence package, so you will have to add the niftyswap binding in go-sequence/contracts. To do this, just update go-sequence/contracts/package.json to add @0xsequence/niftyswap, and follow other conventions like other contracts, and then it will code-generate bindings for you which you can use here too.

Still, I think I am missing a go:generate line, working on that.

Do we need this change here or have I misunderstood things?

File: contracts/gen/gen.go [hunk 1]
```diff
@@ -19,6 +19,16 @@
 //go:generate go run github.com/0xsequence/ethkit/cmd/ethkit abigen --pkg=tokens --type=IERC721 --outFile=./tokens/ierc721.gen.go --artifactsFile=../artifacts/erc721/IERC721.json
 //go:generate go run github.com/0xsequence/ethkit/cmd/ethkit abigen --pkg=tokens --type=IERC1155 --outFile=./tokens/ierc1155.gen.go --artifactsFile=../artifacts/erc1155/interfaces/IERC1155.sol/IERC1155.json
 
+//
```
Comment (xiam) [inline contracts/gen/gen.go:22]: @pkieltyka do parameters look correct here? should we include `INifty*` too?

File: contracts/contracts.go [hunk 2]
```diff
@@ -53,6 +57,10 @@ func init() {
 	IERC721 = artifact("IERC721", tokens.IERC721ABI, "")
 	IERC1155 = artifact("IERC1155", tokens.IERC1155ABI, "")
 
+	NiftyswapExchange = artifact("NIFTYSWAP_EXCHANGE", contracts.NiftyswapExchangeABI, contracts.NiftyswapFactoryBin)
```
Comment (xiam) [inline contracts/contracts.go:60]: @pkieltyka I'm also unsure about this part, I tried to make it look similar to the others, does it look correct? Should we also use or add `INifty*` stuff?

**Comment by xiam** [comment]
I'm going to mark this as a draft and will remove the status once https://github.com/horizon-games/issue-tracker/issues/5306 is completed.

**Review by pkieltyka — APPROVED** [review]

**Pull request merged**